File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
    if (needle.length === 0) return 0;
    let i = 0
    let j = 0
    while (i < haystack.length) {
        if (haystack[i] === needle[j]) {
            i++;
            j++;
            if(j === needle.length){
                return i - j;
            }
        }
        else{
            i = i - j + 1
            j = 0
        }
    }
    return -1;
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if len(needle) == 0:
            return 0
        
        i = 0
        j = 0
        while i < len(haystack):
            if haystack[i] == needle[j]:
                i = i + 1
                j = j + 1
                if j == len(needle):
                    return i - j
            else:
                i = i - j + 1
                j = 0
        return -1
            
        